### PR TITLE
Added Lens formulae to the Physics repository

### PR DESCRIPTION
**Pull request by san-jay-14**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: physics/lens_formulae.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    if image_distance_from_lens == 0 or focal_length_of_lens == 0:
+        raise ValueError(
+            "Invalid inputs. Enter non zero values with respect to the sign convention."
+        )
+
+    object_distance = 1 / ((1 / image_distance_from_lens) - (1 / focal_length_of_lens))
+    return object_distance
+
+
+def image_distance(focal_length_of_lens: float, object_distance_from_lens: float):
```
[the remark this answers is not in the document]
Comment (san-jay-14) [reply physics/lens_formulae.py:109]: Sorry, I will review and resolve in few hours.

**Comment by san-jay-14** [comment]
I have Resolved all the issues commented in my code. Kindly go through my code and if any issues found comment them  and I'll look into it and try to resolve as soon as possible.

**Comment by san-jay-14** [comment]
@cclauss Can you please take a look into my PR.
Thank you in advance 🙂

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
Please remove the indentation in the header docstring so that there's more space for text

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**